_ml-commons-plugin/agents-tools/tools/dynamic-tool.md

```diff
@@ -0,0 +1,182 @@
+POST /_plugins/_ml/agents/9X7xWI0Bpc3sThaJdY9i/_execute
+{
+  "parameters": {
+    "question": "How many active primary shards are there in the OpenSearch cluster?"
+  }
+}
+```
+{% include copy-curl.html %} 
+
+OpenSearch returns the `ClusterHealthTool` results:
+
+```json
+{
+    "inference_results": [
+        {
+            "output": [
+                {
+                    "name": "response",
+                    "result": "7"
+                }
+            ]
+        }
+    ]
+}
+```
+
+## Running Node Stats API as a tool
+
+To run [Node Stats API]({{site.url}}{{site.baseurl}}/api-reference/nodes-apis/nodes-stats/) as a tool, follow these steps.
+
+### Step 1: Register a flow agent that will run the DynamicTool
+
+To create a flow agent, send the following register agent request, specifying the tool's parameters:
+
+```json
+POST /_plugins/_ml/agents/_register
+{
+  "name": "Test_Agent_For_DynamicTool",
+  "type": "flow",
+  "description": "this is a test agent that converts node stats API to a tool via DynamicTool",
+  "tools": [
+    {
+      "type": "DynamicTool",
+      "name": "NodeStatsTool",
+      "parameters": {
+        "method": "GET",
+        "uri": "/_nodes/${node_id}/stats",
+        "response_filter": "$.thread_pool.index_searcher"
+      }
+    }
+  ]
+}
+```
+{% include copy-curl.html %} 
+
+For parameter descriptions, see [Register parameters](#register-parameters).
+
+OpenSearch responds with an agent ID:
+
+```json
+{
+  "agent_id": "9X7xWI0Bpc3sThaJdY9i"
+}
+```
+
+### Step 2: Run the agent
+
+Then, run the agent by sending the following request:
+
+```json
+POST /_plugins/_ml/agents/9X7xWI0Bpc3sThaJdY9i/_execute
+{
+  "parameters": {
+    "question": "What are the index searcher thread pool stats in the OpenSearch cluster?",
+    "node_id": "-bi3Rw1OTA6-jK4Uh5_Thg"
+  }
+}
+```
+{% include copy-curl.html %} 
+
+OpenSearch returns the `NodeStatsTool` results:
+
+```json
+{
+    "inference_results": [
+        {
+            "output": [
+                {
+                    "name": "response",
+                    "result": "{\"threads\": 0,\"queue\": 0,\"active\": 0,\"rejected\": 0,\"largest\": 0,\"completed\": 0,\"total_wait_time_in_nanos\": 0}"
+                }
+            ]
+        }
+    ]
+}
+```
+
+
+## Register parameters
+
+The following table lists all tool parameters that are available when registering an agent.
+
+
+| Parameter | Type | Required/Optional | Description |
+|:---|:---|:---|:---|
+| `method` | String | Required | The HTTP method used by the API (for example, `GET`, `POST`).                                                                               |
+| `uri` | String | Required | The API URI. Supports `${placeholder}` syntax. Placeholders are replaced at runtime with values from the agent request body. |
+| `request_body`    | String | Optional | The API request body. Supports `{placeholder}` syntax. Placeholders are replaced at runtime with values from the agent request body. |
```

Comment:
@zane-neo Should this be `${placeholder}`?

Reply:
It should be Placeholders since it refers to all the `${placeholder}` mentioned before.

Reply:
Fixed.